airbyte-metrics/metrics-lib/src/main/java/io/airbyte/metrics/lib/ApmTraceUtils.java

```diff
@@ -87,6 +87,21 @@ public static void addExceptionToTrace(final Span span, final Throwable t) {
     }
   }
 
+  /**
+   * Adds all the provided tags to the root span.
+   *
+   * @param tags A map of tags to be added to the root span.
+   */
+  public static void addTagsToRootSpan(final Map<String, Object> tags) {
+    final Span activeSpan = GlobalTracer.get().activeSpan();
+    if (activeSpan instanceof MutableSpan) {
```

Comment:
[question] If the framework is wrapping all requests in a span, how would we _not_ have a span?  Perhaps if Data dog is disabled?

Reply:
@evantahler Good catch.  This method should check if `activeSpan` is not `null`.  It will be `null` when the DataDog Java agent is not present.  cc:  @alafanechere

Reply:
Thanks for the suggestion. Isn't `if activeSpan instanceof MutableSpan` already checking a `not null`  by itself?

Reply:
@evantahler `MutableSpan` is a datadog interface. And `Span` and  `GlobalTracer.get().activeSpan()` are from `opentelemetry`. So when datadog is disabled we might not received Span object implementing the MutableSpan interface, hence the importance of this check. Am I right @jdpgrailsdev ?

Reply:
@alafanechere Correct on both accounts.  The `instanceof` check does protect us from `null`, as it will resolve to `false`.  `MutableSpan` is a DataDog only class, so if a customer uses a different APM product that supports OpenTracing, that check protects us from calling methods that don't exist on the object.